### ISSUE TITLE
Running `toolbox enter` for every terminal prompt will get annoying quickly

### ISSUE THREAD

**Issue by allanday** [containers/toolbox#135]
On Silverblue, if we expect users to generally be using a toolbox as their terminal environment, then having to run `toolbox enter` every time you open a new terminal will get annoying really quickly.

Potential solutions to this:

 1. Show separate terminal launchers - one for the host, one for each toolbox
 2. Have the terminal open in the last used environment (could be a toolbox or the host); one issue here: how you then get from the toolbox container to the host

From a pure UX perspective I think I prefer the second option, because it puts less stress on the user: with the first option, you have to think about where you want to open the terminal and pick the right launcher (potentially from a list), which is harder work than just launching the terminal it generally doing the right thing.

**Comment by debarshiray** [comment]
>     2. Have the terminal open in the last used toolbox (one issue here: how you then
> get from the toolbox container to the host)

We could offer a command to open a shell on the host. Could be a `toolbox` command or something separate.

**Comment by allanday** [comment]
> We could offer a command to open a shell on the host. Could be a `toolbox` command or something separate.

That could be cool.

**Comment by allanday** [comment]
Another issue with option 1: someone has to learn to use the toolbox terminal. Their first experience is likely be to try what they know, which will be to run the standard terminal app.

There is another option of course:

 3. Have one terminal, and make it default to a toolbox environment

The obvious issue with this is that it will be painful for anyone who needs to use the host environment. There's a bit of an unknown here: we're expecting most people to use the toolbox most of the time, but there might be cases we don't know about where people need the host.

**Comment by debarshiray** [comment]
https://gitlab.freedesktop.org/terminal-wg/specifications/issues/17 is meant to reduce the need to keep typing `toolbox enter` over and over again. The idea is that you'd run `toolbox enter` once, and the terminal will take care of spawning the right environment from there onwards.

Related to that is this GNOME Terminal issue about retaining the current working directory for commands other than the host shell: https://gitlab.gnome.org/GNOME/gnome-terminal/issues/126

**Comment by debarshiray** [comment]
Part of the solution was implemented in pull request #199 When used with [matching](https://bodhi.fedoraproject.org/updates/FEDORA-2019-5aca07c69c) GNOME Terminal and VTE versions, the toolbox container will be preserved when opening a new terminal.

There's still no `toolbox` command to request a host shell from inside the container, though.

**Comment by kamalmarhubi** [comment]
Near as I can tell, after #199 there is no way to open a host terminal unless you have one lying around somewhere. Is this correct?

**Comment by basvdlei** [comment]
> Near as I can tell, after #199 there is no way to open a host terminal unless you have one lying around somewhere. Is this correct?

I don't think so. But you can start a new terminal (eg. by holding the `Ctrl` key when clicking the terminal icon in Gnome).

@debarshiray would you consider adding the option to disable this feature with something like a gsetting, enviroment variable or dotfile? Most of my work consists of starting containers which is not really practical to do from inside the toolbox.

**Comment by rugk** [comment]
Opened https://github.com/debarshiray/toolbox/issues/218 for disabling entering the toolbox. IMHO this issue here is fixed, is not it?

**Comment by mildred** [comment]
For the records, implementation in gnome-terminal is:

- https://src.fedoraproject.org/rpms/gnome-terminal/c/150fa9196e8e1ecda5439915c086c3f98cf51a1d?branch=master
- https://src.fedoraproject.org/rpms/vte291/c/3a1a7ca5f75e6b57717f748757a534a7d75301e2?branch=master

**Comment by mildred** [comment]
Escape sequence spec issue: https://gitlab.freedesktop.org/terminal-wg/specifications/issues/17

**Comment by sebhoss** [comment]
I'm typically working on multiple projects at the same time which all require different tooling and different versions of those tools. So I've used [direnv](https://direnv.net/) to call `toolbox enter` automatically as soon as I `cd` into those projects like this:

1) Edit `~/.direnvrc` and add
```
use_toolbox() {
  local name=${1:-fedora-toolbox-30}
  echo using toolbox: $name
  toolbox --assumeyes enter --container $name
}
```
2) Edit `<path-to-project>/.envrc` to just
```
use toolbox
```

and once I `cd path-to-project` I'm automatically in a toolbox container. You can specify a different container with something like `use toolbox <container-name>` in the `.envrc` file.

**Comment by ljrk0** [comment]
While this issue is solved for using key combinations, for automation this isn't yet workable.

1. Install `gnome-terminal`  *inside* the toolbox
2. Run `gnome-terminal`
3. Observe that while the working directory is preserved, the shell terminals shell is outside of the toolbox.

This would be useful for all kinds of CLI apps that programmatically launch a second terminal window to display other output (e.g., the kakoune editor in IDE mode).

However, maybe it would be better to solve this on the gnome terminal side and allow some IPC way for the shell to notify its connected terminal emulator to spawn a new instance, not unlike kitty does.  This would, I think, solve many issues w/o needing new escape sequences and perhaps even making it easier for those who don't want this feature (https://github.com/containers/toolbox/issues/218) to opt-out.

Should this issue be re-opened or should I open a new one, referring to this, or open one over at gnome terminal?
